File: sdk/examples/quickstart.ts
import { createToken, checkScope, generateKeyPair } from '../src/index.js';

const keys = await generateKeyPair();

const token = await createToken({
  principal: { id: "did:apoa:you" },
  agent: { id: "did:apoa:your-agent", name: "My Agent" },
  services: [{
    service: "mychart.com",
    scopes: ["appointments:read", "prescriptions:read"],
    constraints: { signing: false, data_export: false }
  }],
  expires: "2026-09-01"
}, { privateKey: keys.privateKey });

// Can the agent read appointments? Yes.
const allowed = checkScope(token, "mychart.com", "appointments:read");
console.log("appointments:read →", allowed);
// { allowed: true, reason: "matched scope 'appointments:read'", matchedScope: "appointments:read" }

// Can the agent send messages? Absolutely not.
const denied = checkScope(token, "mychart.com", "messages:send");
console.log("messages:send →", denied);
// { allowed: false, reason: "scope 'messages:send' not in authorized scopes" }

File: sdk-python/examples/quickstart.py
"""Minimal example: create a token, check what the agent can and can't do."""

from apoa import (
    APOADefinition,
    Agent,
    Principal,
    ServiceAuthorization,
    SigningOptions,
    check_scope,
    create_token,
    generate_key_pair,
)


def main() -> None:
    private_key, _ = generate_key_pair()

    token = create_token(
        APOADefinition(
            principal=Principal(id="did:apoa:you"),
            agent=Agent(id="did:apoa:your-agent", name="My Agent"),
            services=[
                ServiceAuthorization(
                    service="mychart.com",
                    scopes=["appointments:read", "prescriptions:read"],
                    constraints={"signing": False, "data_export": False},
                ),
            ],
            expires="2026-09-01",
        ),
        SigningOptions(private_key=private_key),
    )

    # Can the agent read appointments? Yes.
    allowed = check_scope(token, "mychart.com", "appointments:read")
    print("appointments:read ->", allowed)

    # Can the agent send messages? Absolutely not.
    denied = check_scope(token, "mychart.com", "messages:send")
    print("messages:send ->", denied)


if __name__ == "__main__":
    main()
